### PR TITLE
Combine `const` and `enum` errors into a single message

### PR DESCRIPTION
**Pull request by ShivaGupta-14**
### Description

When multiple `const` or `enum` constraints fail (e.g., via `allOf`), produce a single message with the most restrictive constraint instead of multiple redundant messages.

- Compute the intersection of allowed values across all constraints
- If intersection has 1 value: use [const](cci:1://file:///Users/sst/Desktop/JSON/json-schema-errors/src/json-schema-errors.js:39:0-68:2) message
- If intersection has multiple values: use `enum` message
- If intersection is empty (contradictory): use boolean schema message
- Fixes: #132

### Checklist

- [x] npm test
- [x] npm run lint
- [x] npm run type-check

### CODE REVIEW

**Comment by ShivaGupta-14** [comment]
> This approach isn't going to work. Try,
> 
> ```json
> {
>   "allOf": [
>     { "const": "a" },
>     { "const": "b" }
>   ]
> }
> ```

Hi @jdesrosiers,

thanks for pointing out the issue! I understood the problem:

**Issue**: with a schema like above one -> `allOf: [{ const: "a" }, { const: "b" }]`, if the instance is `"a"`

- const: "a" passes (not in errors)
- Only const: "b" fails
- My original code only saw failing constraints -> couldn't detect the contradiction
- My Solution: I updated the handler to collect ALL constraints (both passing and failing), then compute the intersection:

```jsx
const passed = normalizedErrors[...][schemaLocation] === true;
if (!passed) hasFailure = true;
// always collect
constraints.push({ allowedValues: [...], schemaLocation });
```

**now**:
- Both ["a"] and ["b"] are collected
- Intersection = [] (empty)
- Returns getBooleanSchemaErrorMessage()
 
Tests Added:

- contradictory const - empty intersection (instance matches one const)
- contradictory enum - empty intersection (disjoint enum sets)

All tests pass. Is this approach good, or does it need further improvements?
Thanks!

File: src/error-handlers/constAndEnum.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    a.allowedValues.length - b.allowedValues.length
+  );
+  const mostConstraining = sorted[0];
+
+  let intersection = mostConstraining.allowedValues;
+  for (let i = 1; i < sorted.length; i++) {
+    intersection = intersection.filter((/** @type {Json} */ val) =>
+      sorted[i].allowedValues.some((/** @type {Json} */ other) => jsonEqual(val, other))
+    );
+  }
```
Comment (jdesrosiers) [inline src/error-handlers/constAndEnum.js]: All of this could be avoided if you used a Set and Set operations to determine the intersection.

Comment (ShivaGupta-14) [reply src/error-handlers/constAndEnum.js]: yes, I looked into `Set.intersection()` but didn't use it due to compatibility concerns, it requires Node 22+, Chrome 122+, Safari 17+

The manual approach ensures broader compatibility. should I switch to Set.intersection(), or is the manual approach preferred for this project?

Comment (jdesrosiers) [reply src/error-handlers/constAndEnum.js]: Yes. Use `Set.intersection`.

**Comment by ShivaGupta-14** [comment]
> It says, "Expected one of ⁨"a" or "b"⁩" with schema location `/allOf/0/enum`, but that's a passing location. If we exclude passing locations, we get a schema location of `/allOf/1/enum`, which is better, but it seems contradictory because the message says only "a" and "b" are allowed, but the schema location says that "d" is also allowed. That means we need to include all the failing schema locations to make it clear why only "a" and "b" are allowed. I think the only time we can collapse to a single most constraining location is when the effective allowed set matches exactly one of the keywords.

Thanks for pointing it out, I missed it. for such cases, I’ve updated the logic.

- If the intersection matches exactly one constraint's values -> use that single location
- Otherwise -> include all constraint locations

this will handle edge case: when intersection is ["a","b"] but no single constraint has exactly ["a","b"], we show all locations so users understand why only "a" and "b" are valid.

also, test added for this case: "multiple enums with no exact match"

File: src/error-handlers/constAndEnum.js (hunk 1 above)
Comment (ShivaGupta-14) [reply src/error-handlers/constAndEnum.js]: Got it, makes sense! thanks for explaining!
